pkg/windows/calico.go

```diff
@@ -302,7 +302,8 @@ func (c *Calico) Start(ctx context.Context) error {
 
 // generateCalicoNetworks creates the overlay networks for internode networking
 func (c *Calico) generateCalicoNetworks() error {
-	if err := deleteAllNetworks(); err != nil {
+	cmd := exec.Command("powershell.exe", "C:\\var\\lib\\rancher\\rke2\\bin\\node-calico.ps1")
```

Comment:
We have in the past pushed back on just having golang call out to powershell to do things. If we wanted to do this, we could just use upstream's powershell scripts in the first place, instead of handling cni setup in golang. Why do we need a powershell script to check a registry key and reinitialize hns?

Reply:
It's need to check the latest reboot. If Calico os starting after a reboot it has to clean the network if it's after a restart of RKE2 the network shouldn't be cleaned. I can try to do the check inside golang.

Reply:
Updated